astropy/coordinates/representation.py

```diff
@@ -1564,6 +1913,23 @@ def _check_base(cls, base):
                             '{0}, not {1}.'.format(cls.base_representation,
                                                    type(base)))
 
+    def _get_deriv_key(self, base):
+        """Given a base (representation instance), determine the unit of the
+        derivative by removing the representation unit from the component units
+        of this differential.
+        """
+
+        # TODO: better validation by checking, e.g., _compatible_differentials
+        # for this class
+        self._check_base(base)
```

Comment:
I think this may indeed be too strict, but we can revisit if it turns out to be a problem.